server/src/main/java/org/elasticsearch/cluster/ClusterModule.java

```diff
@@ -337,6 +337,10 @@ private static ShardsAllocator createShardsAllocator(
         return Objects.requireNonNull(allocatorSupplier.get(), "ShardsAllocator factory for [" + allocatorName + "] returned null");
     }
 
+    public AllocationDeciders getAllocationDeciders() {
+        return allocationDeciders;
+    }
```

Comment:
Rather than add this at the `ClusterModule` level, we could add it to the actual `AllocationService` (since it has access to the deciders internally in the first place). That might make more sense since the deciders are more part of the allocation service than the cluster module; what do you think?

Reply:
Fixed in [3819d46](https://github.com/elastic/elasticsearch/pull/85071/commits/3819d460d24a726a25bb8e27880522065b95c21a)